ansible/library/conn_graph_facts.py

```diff
@@ -233,29 +233,49 @@ def main():
     module = AnsibleModule(
         argument_spec=dict(
             host=dict(required=False),
+            hosts=dict(required=False, type='list'),
             filename=dict(required=False),
         ),
+        mutually_exclusive=[['host', 'hosts']],
         supports_check_mode=True
     )
     m_args = module.params
-    hostname = m_args['host']
+
+    hostnames = m_args['hosts']
+    if not hostnames:
+        hostnames = [m_args['host']]
     try:
         if m_args['filename']:
             filename = m_args['filename']
         else:
             filename = LAB_GRAPHFILE_PATH + LAB_CONNECTION_GRAPH_FILE
         lab_graph = Parse_Lab_Graph(filename)
         lab_graph.parse_graph()
-        dev = lab_graph.get_host_device_info(hostname)
-        if dev is None:
-            module.fail_json(msg="cannot find info for "+hostname)
-        results = {}
-        results['device_info'] =  lab_graph.get_host_device_info(hostname)
-        results['device_conn'] = lab_graph.get_host_connections(hostname)
-        if lab_graph.get_host_vlan(hostname):
-            results['device_vlan_range'] = lab_graph.get_host_vlan(hostname)['VlanRange']
-            results['device_vlan_list'] = lab_graph.get_host_vlan(hostname)['VlanList']
-        results['device_port_vlans'] = lab_graph.get_host_port_vlans(hostname)
+
+        device_info = []
+        device_conn = []
+        device_port_vlans = []
+        device_vlan_range = []
+        device_vlan_list = []
+        for hostname in hostnames:
+            dev = lab_graph.get_host_device_info(hostname)
+            if dev is None:
+                module.fail_json(msg="cannot find info for %s" % hostname)
+            device_info.append(dev)
+            device_conn.append(lab_graph.get_host_connections(hostname))
+            host_vlan = lab_graph.get_host_vlan(hostname)
+            # for spytest multi-DUTs, must ensure all have vlan configured.
+            if host_vlan:
+                device_vlan_range.append(host_vlan["VlanRange"])
+                device_vlan_list.append(host_vlan["VlanList"])
+            device_port_vlans.append(lab_graph.get_host_port_vlans(hostname))
+        results = {k: v for k, v in locals().items()
+                   if (k.startswith("device_") and v)}
+
+        # flatten the lists for single host
+        if len(hostnames) == 1:
+            results = {k: v[0] for k, v in results.items()}
```

Comment:
I think it would be better to create another fixture explicitly to return the only graph instead of making it special. I also think the multiple DUT graph should have a different name. If all tests are getting graph with the same fixture. It would be hard to transition from the old fixture to the new fixture.